api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsExpirationServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package org.pmiops.workbench.initialcredits;
+
+import java.sql.Timestamp;
+import java.util.Optional;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbUserInitialCreditsExpiration;
+import org.springframework.stereotype.Service;
+
+@Service
+public class InitialCreditsExpirationServiceImpl implements InitialCreditsExpirationService {
+
+  @Override
+  public Optional<Timestamp> getCreditsExpiration(DbUser user) {
+    return Optional.ofNullable(user.getUserInitialCreditsExpiration())
+        .filter(exp -> !exp.isBypassed()) // If the expiration is bypassed, return empty.
+        // TODO filter on institutional bypass as well, maybe something like
```

Comment:
doesn't exist yet.  I also didn't see a story for this, so I added a note in the design doc.